bfs_battle_for_supremacy/game_logic/managers/map_manager.py

```diff
@@ -1,9 +1,135 @@
 from bfs_battle_for_supremacy.game_logic.entities.map import Map
 from bfs_battle_for_supremacy.game_logic.entities.square import Square
+from collections import deque
 
 
 class MapManager:
     game_map = Map(10, 10)
+    movement_counter = 20
+
+    current_position = None
+    current_object = None
+    target_position = None
+    selection_counter = 0
+
+    @staticmethod
+    def reset_movement_counter():
+        MapManager.movement_counter = 20
+
+    @staticmethod
+    def reset_selection():
+        MapManager.current_position = None
+        MapManager.current_object = None
+        MapManager.target_position = None
+        MapManager.selection_counter = 0
+
+    @staticmethod
+    def select_square(square: Square, current_player):
+        if MapManager.selection_counter == 0:
+            if not square.is_empty:
+                content = square.get_content()
+                if (
+                    content == current_player
+                    or content in current_player.monsters
+                ):
+                    MapManager.current_position = square
+                    MapManager.current_object = content
+                    MapManager.selection_counter = 1
+                else:
+                    print("Invalid selection. Please select a valid object.")
+            else:
+                print("Invalid selection. Square is empty.")
+
+        elif MapManager.selection_counter == 1:
```

Comment:
here we check if the player previously selected something movable (the player or his monsters) then apply the move.

else it should reselect the current square and remove the old selecting